File: JavaAndDotnet/Dotnet/labpt10/labpt8/GeneratorDanych.cs
using System;
using System.IO;

namespace ExampleApp
{
    /// <summary>
    /// Generates a CSV file named "data.csv" with 10 000 000 rows.
    /// Format per row: ID,FirstName,Hillar,Age,BoneFractures
    /// (male first names chosen uniformly at random, age 18–65, boneFractures 6–12).
    /// </summary>
    public static class GeneratorDanych
    {
        private static readonly string[] MaleNames = new[]
        {
            "Adam", "Adrian", "Aleksander", "Andrzej", "Antoni", "Artur", "Bartlomiej", "Beniamin",
            "Blazej", "Cezary", "Damian", "Daniel", "Dawid", "Dominik", "Edward", "Eryk", "Fabian", "Filip",
            "Franciszek", "Gabriel", "Grzegorz", "Henryk", "Hubert", "Igor", "Jacek", "Jakub", "Jan", "Janusz",
            "Jaroslaw", "Jonasz", "Jozef", "Julian", "Kacper", "Kajetan", "Karol", "Kazimierz", "Konrad", "Krystian",
            "Ksawery", "Lukasz", "Maciej", "Maksymilian", "Marcel", "Marcin", "Marek", "Mateusz", "Michal", "Mikolaj",
            "Miroslaw", "Mariusz", "Norbert", "Oskar", "Patryk", "Piotr", "Przemyslaw", "Radoslaw", "Robert", "Sebastian",
            "Stanislaw", "Szymon", "Tadeusz", "Tomasz", "Witold", "Wojciech", "Zbigniew", "Zdzislaw", "Zygmunt"
        };

        public static void GenerateData()
        {
            const string csvFile = "data.csv";
            const int totalRows = 100_000;
            var rand = new Random();

            using (var writer = new StreamWriter(csvFile))
            {
                // Write header (optional; Java version did not write a header)
                // writer.WriteLine("ID,FirstName,LastName,Age,BoneFractures");

                for (int i = 0; i < totalRows/2; i++)
                {
                    // Age: 18–65
                    int age = rand.Next(18, 66);
                    // Bone fractures: 6–12
                    int boneFractures = rand.Next(6, 13);
                    // Random male name from array
                    string firstName = MaleNames[rand.Next(MaleNames.Length)];
                    // Last name is always "Hillar"
                    writer.WriteLine($"{i},{firstName},Hillar,{age},{boneFractures}");
                }
                for (int i = totalRows / 2; i < totalRows; i++)
                {
                    // Age: 18–76
                    int age = rand.Next(18, 76);
                    // Bone fractures: 2-6
                    int boneFractures = rand.Next(2, 6);
                    // Random male name from array
                    string firstName = MaleNames[rand.Next(MaleNames.Length)];
                    // Last name is always "Moniewski"
                    writer.WriteLine($"{i},{firstName},Moniewski,{age},{boneFractures}");
                }
            }
        }
    }
}

File: JavaAndDotnet/Java/pt_d3_ga_j_czuba_moniewski_kulesza-8d82d87d4cdec28ceb2625fb5c0733b727bdb1aa/src/main/java/org/example/GeneratorDanych.java
package org.example;

import java.io.FileWriter;
import java.io.IOException;
import java.util.Random;

public class GeneratorDanych {
    public static void main(String[] args) {
        String csvFile = "data.csv";
        String[] maleNames = {
                "Adam", "Adrian", "Aleksander", "Andrzej", "Antoni", "Artur", "Bartlomiej", "Beniamin",
                "Blazej", "Cezary", "Damian", "Daniel", "Dawid", "Dominik", "Edward", "Eryk", "Fabian", "Filip",
                "Franciszek", "Gabriel", "Grzegorz", "Henryk", "Hubert", "Igor", "Jacek", "Jakub", "Jan", "Janusz",
                "Jaroslaw", "Jonasz", "Jozef", "Julian", "Kacper", "Kajetan", "Karol", "Kazimierz", "Konrad", "Krystian",
                "Ksawery", "Kuba", "Lech", "Leon", "Leonard", "Lukasz", "Maciej", "Maksymilian", "Marcel", "Marcin",
                "Mariusz", "Mateusz", "Maurycy", "Michal", "Mieczyslaw", "Miroslaw", "Natan", "Nikodem", "Norbert", "Olaf",
                "Oskar", "Pawel", "Piotr", "Przemyslaw", "Rafal", "Robert", "Roman", "Ryszard", "Sebastian", "Seweryn",
                "Slawomir", "Stanislaw", "Stefan", "Szymon", "Tadeusz", "Tomasz", "Wieslaw", "Wiktor", "Wincenty", "Witold",
                "Wladyslaw", "Zbigniew", "Zdzislaw"
        };
        int length = maleNames.length;
        try (FileWriter writer = new FileWriter(csvFile)) {
            writer.append("ID,Name,Surname,Age,BoneFractures\n");
            int indexName=0;
            Random random = new Random();
            int age=random.nextInt(81) + 20;
            int boneFractures=0;
            int i;
            for (i = 0; i < 20000000; i++) {
                age=random.nextInt(81) + 20;
                boneFractures=random.nextInt(15) + 3;
                String name=maleNames[random.nextInt(length)];
                writer.append(i + "," + name + ",Czuba," + age + "," + boneFractures + "\n");
            }
            for (i = 2000000; i < 40000000; i++) {
                age=random.nextInt(81) + 20;
                boneFractures=random.nextInt(7) + 6;
                String name=maleNames[random.nextInt(length)];
                writer.append(i + "," + name + ",Jaworski," + age + "," + boneFractures + "\n");
            }
            for (i = 40000000; i < 60000000; i++) {
                age=random.nextInt(81) + 20;
                boneFractures=random.nextInt(7) + 6;
                String name=maleNames[random.nextInt(length)];
                writer.append(i + "," + name + ",Moniewski," + age + "," + boneFractures + "\n");
            }
            for (i = 60000000; i < 80000000; i++) {
                age=random.nextInt(81) + 20;
                boneFractures=random.nextInt(7) + 6;
                String name=maleNames[random.nextInt(length)];
                writer.append(i + "," + name + ",Kulesza," + age + "," + boneFractures + "\n");
            }
            for (i = 80000000; i < 100000000; i++) {
                age=random.nextInt(81) + 20;
                boneFractures=random.nextInt(7) + 6;
                String name=maleNames[random.nextInt(length)];
                writer.append(i + "," + name + ",Hillar," + age + "," + boneFractures + "\n");
            }



        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
